presto-docs/src/main/sphinx/gcp/storage.rst

```diff
@@ -0,0 +1,216 @@
+    </property>
+
+More configuration properties and their default values are shown in `gcs-core-default.xml <https://github.com/GoogleCloudPlatform/bigdata-interop/blob/v1.9.10/gcs/conf/gcs-core-default.xml>`_.
+
+Hive Connector
+^^^^^^^^^^^^^^
+
+Another requirement is that you have enabled and configured a Hive connector in Presto.
+The connector uses Hive metastore for data discovery and is not limited to data residing on HDFS.
+
+**Configuring Hive Connector**
+
+* URL to Hive metastore:
+
+    * New Hive metastore on GCP:
+
+        If your Presto nodes are provisioned by GCP, your Hive metastore should also be on GCP to minimize latency and costs.
+        The simplest way to create a new Hive metastore on GCP is to create a small Cloud DataProc cluster (1 master, 0 workers), accessible from your Presto cluster.
+        Follow the steps for existing Hive metastore after finishing this step.
+
+    * Existing Hive metastore:
+
+        To use an existing Hive metastore with a Presto cluster, you need to set the ``hive.metastore.uri`` property in your Hive catalog properties file to ``thrift://${METASTORE_ADDRESS}:${METASTORE_THRIFT_PORT}``.
+        If the metastore uses authentication, please refer to `Hive Security Configuration`_.
+
+        .. _Hive Security Configuration: https://docs.starburstdata.com/latest/connector/hive-security.html
+
+* GCS access:
+
+    * Simplified configuration:
+
+        Here are example values for all GCS configuration properties which can be set in the Hive catalog properties file:
+
+        .. code-block:: properties
+
+            # JSON key file used to access Google Cloud Storage
+            hive.gcs.json-key-file-path=/path/to/gcs_keyfile.json
+
+            # Use client-provided OAuth token to access Google Cloud Storage
+            hive.gcs.use-access-token=false
+
+    * Full configuration:
+
+        If you need more configuration options, finish the `Full GCS configuration`_ section and then add the full path to your ``core-site.xml`` file to the ``hive.config.resources`` property in the Hive catalog properties file.
+        This will allow Presto Hive connector to pick up GCS configuration during startup.
+
+Accessing GCS Data From Presto for the First Time
+-------------------------------------------------
+
+Accessing Data Already Mapped in the Hive metastore
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you migrate to Presto from Hive, chances are that your GCS data is already mapped to SQL tables in the metastore. 
+In that case, you should be able to query it.
+
+Accessing Data Not Yet Mapped in the Hive metastore
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To access GCS data that is not yet mapped in the Hive metastore you need to provide the schema of the data, the file format, and the data location. 
+For example, if you have ORC or Parquet files in an GCS bucket ``my_bucket``, you will need to execute a command similar to the following.
+
+.. code-block:: RST
```

Comment:
For all of these SQL examples, I think it's better to make the code block the format `sql` and make it just the SQL text, so that users can copy/paste, as it is done in the Hive connector docs. In cases where we want to show the output of a command, that can be split into a separate code block, as is done in the `SHOW CREATE TABLE` docs.

Note that plain text like this should use the `none` code block format.

Reply:
Also, SQL is the default language for the docs, so you can use the `::` shorthand to end a sentence with a colon and start a SQL block:
```
Run a query::

    SELECT abc FROM xyz
```
is equivalent to
```
Run a query:

.. code-block:: sql

    SELECT abc FROM xyz
```